lib/App/MintTag.pm

```diff
@@ -105,6 +105,36 @@ sub mint_tag ($self, $auto_mode = 0) {
   };
 }
 
+# Tag an arbitrary sha according to the config file. This bails unless there is
+# exactly one step in config with a defined tag_prefix.
+sub tag_arbitrary_sha ($self, $sha) {
+  my $num_prefixes = 0;
+  my $tagging_step;
+
+  for my $step ($self->config->steps) {
+    if ($step->tag_prefix) {
+      $num_prefixes++;
+      $tagging_step = $step;
+    }
+  }
+
+  unless ($num_prefixes == 1) {
+    die "refusing to tag sha; config file is ambiguous about how to tag.\n";
+  }
+
+  # Now, we can go. The plan is: set up, check out the tag, then tag it and
+  # push it.
+  $self->prepare_local_directory;
+
+  run_git('checkout', $sha);
+  run_git('submodule', 'update');
```

Comment:
So, maybe_tag_commit looks at rev-parse HEAD, which means it needs to be checked out.  This is fine.  But is it necessary?  If not, probably faster to not do that.  I'm more curious as to why it works like this than annoyed it might not be blazing fast, since it'll never be blazing fast.

Reply:
It works that way mostly because it was expedient in the normal case, which works like:

1. reset to known state
2. approve and merge
3. tag and push

After step 2, HEAD is at the thing we want to tag, so we just tag it and go. We could change that here, by having `maybe_tag_commit` take a sha to tag, but in the normal case we'd have to rev-parse it _anyway_ to get the sha to pass. And as you say, it hardly matters (it would be slightly faster in the uncommon use case), so it was easier not to change it.